pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/parser/ModelParser.groovy

```diff
@@ -1165,7 +1165,16 @@ class ModelParser implements Parser {
      */
     protected ModelASTValue parseArgument(Expression e) {
         if (e instanceof ConstantExpression) {
-            return ModelASTValue.fromConstant(e.value, e)
+            Object val = e.value
+            if (val instanceof BigDecimal) {
+                val = val.doubleValue()
+            } else if (val instanceof BigInteger) {
+                if (val > Long.MAX_VALUE || val < Long.MIN_VALUE) {
+                    errorCollector.error(ModelASTValue.fromConstant(-1, e), Messages.ModelParser_BigIntegerValue())
```

Comment:
Maybe converting to Double? The current impl is fine for me though

Reply:
Nah, I want to stick with int-family rather than floating point. Anyway, tweaked in latest commit.